pyface/ui/qt4/tasks/advanced_editor_area_pane.py

```diff
@@ -315,7 +316,7 @@ def compare(one, two):
                  (isinstance(child, QtGui.QDockWidget) and
                   (visible_only or not self.tabifiedDockWidgets(child))))):
                 children.append(child)
-        children.sort(cmp=compare)
+        children.sort(key=functools.cmp_to_key(compare))
```

Comment:
Thanks.. Good call! I was in drone mode when patching this to get my app working and didn't think too hard..

Fixed now..